lib/libvcc/vcc_vmod.c

```diff
@@ -106,6 +99,21 @@ vcc_ParseImport(struct vcc *tl)
 	}
 
 	SkipToken(tl, ';');
+	if (osym != NULL) {
+		AN(osym->path);
+		if (strcmp(osym->path, fn) != 0) {
+			VSB_printf(tl->sb, "Warning: Module %.*s imported from different location.\n",
+			    PF(mod));
+			vcc_ErrWhere2(tl, t1, tl->t);
+			VSB_printf(tl->sb, "Previous import was here:\n");
+			vcc_ErrWhere2(tl, osym->def_b, osym->def_e);
+			tl->err = 0; // reset error; this is just a warning
```

Comment:
I can't imagine a scenario where multiple imports of the exact same vmod shared library can have dire consequences.  It might be sloppy to do so, but it's far more likely an attempt to mix multiple template VCL snippets.

That case should probably just be silently ignored.

If the imports have different targets and we pick the first we encounter and assume that they're all the same, we're asking for some serious grief.

The proffered example however is germane.

Isn't the better and non-heisenberg solution to turn vmod_dir into vmod_path ?